### PR TITLE
Nat fix - use default stringBinding

### PR DESCRIPTION
**Pull request by Gchaimke**
https://github.com/guardicore/monkey/issues/675

### CODE REVIEW

**Review by AXsagi — APPROVED** [review]

**Comment by jiangwenkkkk** [comment]
comm on,did it is ok, i want use this, but i don't know is ok

File: impacket/dcerpc/v5/dcomrt.py [hunk 1]
```diff
@@ -1232,25 +1232,33 @@ def is_fqdn(self):
         return False
 
     def connect(self, iid = None):
+        print('SMB connect')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 2]
```diff
@@ -1232,25 +1232,33 @@ def is_fqdn(self):
...
         if (self.__target in INTERFACE.CONNECTIONS) is True:
+            print('SMB Thread satrt')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion

```

File: impacket/dcerpc/v5/dcomrt.py [hunk 3]
```diff
@@ -1232,25 +1232,33 @@ def is_fqdn(self):
...
             if current_thread().name in INTERFACE.CONNECTIONS[self.__target] and \
                             (self.__oxid in INTERFACE.CONNECTIONS[self.__target][current_thread().name]) is True:
+                print('SMB DCE satrt')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 4]
```diff
@@ -1232,25 +1232,33 @@ def is_fqdn(self):
...
                 dce = INTERFACE.CONNECTIONS[self.__target][current_thread().name][self.__oxid]['dce']
                 currentBinding = INTERFACE.CONNECTIONS[self.__target][current_thread().name][self.__oxid]['currentBinding']
                 if currentBinding == iid:
                     # We don't need to alter_ctx
                     pass
                 else:
+                    print('SMB NEW DCE satrt')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 5]
```diff
@@ -1232,25 +1232,33 @@ def is_fqdn(self):
...
                     newDce = dce.alter_ctx(iid)
                     INTERFACE.CONNECTIONS[self.__target][current_thread().name][self.__oxid]['dce'] = newDce
                     INTERFACE.CONNECTIONS[self.__target][current_thread().name][self.__oxid]['currentBinding'] = iid
             else:
                 stringBindings = self.get_cinstance().get_string_bindings()
+                print(f'SMB stringBindings: {stringBindings}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 6]
```diff
@@ -1232,25 +1232,33 @@ def is_fqdn(self):
...
                 # No OXID present, we should create a new connection and store it
                 stringBinding = None
                 isTargetFQDN = self.is_fqdn()
+                print(f'SMB isTargetFQDN: {isTargetFQDN}')
+                print(f'SMB isTargetFQDN: {self.get_target()}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 7]
```diff
@@ -1232,25 +1232,33 @@ def is_fqdn(self):
...
                 LOG.debug('Target system is %s and isFQDN is %s' % (self.get_target(), isTargetFQDN))
                 for strBinding in stringBindings:
+                    print(f'SMB strBinding: {strBinding}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 8]
```diff
@@ -1259,16 +1267,20 @@ def connect(self, iid = None):
                     # Does this match exactly what get_target() returns?
                     LOG.debug('StringBinding: %s' % strBinding['aNetworkAddr'])
                     if strBinding['wTowerId'] == 7:
+                        print(f'SMB strBinding: {strBinding["aNetworkAddr"]}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 9]
```diff
@@ -1259,16 +1267,20 @@ def connect(self, iid = None):
...
                         # If there's port information, let's strip it for now.
                         if strBinding['aNetworkAddr'].find('[') >= 0:
                             binding, _, bindingPort = strBinding['aNetworkAddr'].partition('[')
                             bindingPort = '[' + bindingPort
+                            print(f'SMB strBinding port: {bindingPort}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 10]
```diff
@@ -1259,16 +1267,20 @@ def connect(self, iid = None):
...
                         else:
                             binding = strBinding['aNetworkAddr']
                             bindingPort = ''
+                            print(f'SMB strBinding without port: {binding}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 11]
```diff
@@ -1259,16 +1267,20 @@ def connect(self, iid = None):
...
 
                         if binding.upper().find(self.get_target().upper()) >= 0:
                             stringBinding = 'ncacn_ip_tcp:' + strBinding['aNetworkAddr'][:-1]
+                            print(f'SMB stringBinding break: {stringBinding}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 12]
```diff
@@ -1278,12 +1290,16 @@ def connect(self, iid = None):
                             # aNetworkAddr is usually the NetBIOS name and unless you have your DNS resolver
                             # with the right suffixes it will probably not resolve right.
                             stringBinding = 'ncacn_ip_tcp:%s%s' % (self.get_target(), bindingPort)
+                            print(f'SMB stringBinding: {stringBinding}')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 13]
```diff
@@ -1278,12 +1290,16 @@ def connect(self, iid = None):
...
                             break
 
                 LOG.debug('StringBinding chosen: %s' % stringBinding)
                 if stringBinding is None:
                     # Something wen't wrong, let's just report it
-                    raise Exception('Can\'t find a valid stringBinding to connect')
+                    stringBinding = 'ncacn_ip_tcp:%s%s' % (self.get_target(), bindingPort)
+                    print(stringBinding)
+                
+                print(stringBinding)
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 14]
```diff
@@ -1295,6 +1311,7 @@ def connect(self, iid = None):
                 dce = dcomInterface.get_dce_rpc()
 
                 if iid is None:
+                    print('raise IID is None')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 15]
```diff
@@ -1303,19 +1320,22 @@ def connect(self, iid = None):
                 dce.connect()
 
                 if iid is None:
+                    print('raise IID is None')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 16]
```diff
@@ -1303,19 +1320,22 @@ def connect(self, iid = None):
...
                     raise Exception('IID is None')
                 else:
                     dce.bind(iid)
 
                 if self.__oxid is None:
                     #import traceback
                     #traceback.print_stack()
+                    print('raise OXID NONE')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

File: impacket/dcerpc/v5/dcomrt.py [hunk 17]
```diff
@@ -1303,19 +1320,22 @@ def connect(self, iid = None):
...
                     raise Exception("OXID NONE, something wrong!!!")
 
                 INTERFACE.CONNECTIONS[self.__target][current_thread().name] = {}
                 INTERFACE.CONNECTIONS[self.__target][current_thread().name][self.__oxid] = {}
                 INTERFACE.CONNECTIONS[self.__target][current_thread().name][self.__oxid]['dce'] = dce
                 INTERFACE.CONNECTIONS[self.__target][current_thread().name][self.__oxid]['currentBinding'] = iid
+                print('END SMB CONNECTION')
```
Comment (Gchaimke) [inline impacket/dcerpc/v5/dcomrt.py]:
```suggestion
```

**Review by Gchaimke — COMMENTED** [review]
Remove debug prints

**Comment by Gchaimke** [comment]
Tested this solution in local env

**Pull request closed without merging**